pyadjoint/overloaded_type.py

```diff
@@ -320,6 +320,42 @@ def _ad_str(self):
         """
         return str(self)
 
+    def _ad_petsc_vec_read_only(self):
```

Comment:
Should I consider adding an optional argument like with_halos? Or does it not make sense?